ubuntu_image/helpers.py

```diff
@@ -5,8 +5,9 @@
 import logging
 
 from contextlib import ExitStack, contextmanager
+from parted import Device
 from subprocess import PIPE, run as subprocess_run
-from tempfile import TemporaryDirectory
+from tempfile import TemporaryDirectory, mkstemp
```

Comment:
Is there a reason why you're using `mkstemp` instead of `NamedTemporaryFile`?  The latter should work well enough for these purposes and doesn't suffer from leaking file descriptors.  You could solve that problem by `os.close()`ing the 0-th item returned from `mkstemp()` but using `NamedTemporaryFile` in a `with` statement solves that for you, e.g.:

```python
def get_default_sector_size():
    with NamedTemporaryFile() as fp:
        os.truncate(fp.name, 0)
        os.truncate(fp.name, 1024*1024)
        device = Device(fp.name)
        return device.sectorSize
```